### PR TITLE
Docs 2.0

### PR DESCRIPTION
**Pull request by fabioberger**
## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

**Comment by chriskalani** [comment]
WoOo!

**Comment by chriskalani** [comment]
@fabioberger @piotrjanosz - can we get these build errors fixed? Found a way around it so I could work on it, but we should address

File: packages/website/ts/components/docs/sidebar/collapse.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+interface ICollapse {
+    isActive: boolean;
+    height?: number;
+}
+
+export const Collapse: React.FC<ICollapseProps> = props => {
+    const [isActive, setIsActive] = React.useState<boolean>(true);
+    // @ts-ignore
```
Comment (fabioberger) [inline packages/website/ts/components/docs/sidebar/collapse.tsx]: What is this ignoring? We haven't ever had to use this before.

File: packages/website/ts/utils/algolia_helpers.ts [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+function processContentTree(tree: Node[], file: File, indexName: string): void {
+    const modify = modifyChildren(modifier);
+    // We first modify the tree to get slugified ids from headings to all text nodes
+    // This is done to be able to link to a certain section in a doc after clicking a search suggestion
+    modify(tree);
+    // Get all text nodes. I.e. 'heading', 'paragraph', 'list' all can have (nested) child text nodes
+    const textNodes = selectAll('text', tree);
+
+    if (textNodes) {
+        // Combines text nodes that exist on the same line. I.e. if a paragraph contains 7 text nodes it combines them into 1. This makes text snippets in algolia more descriptive.
```
Comment (fabioberger) [inline packages/website/ts/utils/algolia_helpers.ts]: Please wrap this comment.

File: packages/website/webpack/mdx-section-toc.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
+const { toJSX } = require('@mdx-js/mdx/mdx-hast-to-jsx');
```
Comment (fabioberger) [inline packages/website/webpack/mdx-section-toc.js]: Use underscroes for file names.

Comment (fabioberger) [reply packages/website/webpack/mdx-section-toc.js]: Also, what does this file do exactly?

**Pull request merged**